pkg/istio/authorizationpolicy/controller_test.go

```diff
@@ -227,12 +240,20 @@ func TestSyncService(t *testing.T) {
 		},
 		{
 			name:                "create Authorization Policy spec when there is update event of service from no annotation set to annotation set",
-			inputService:        notOnboardedServiceWithAnnotation,
+			inputService:        notOnboardedServiceWithAnnotationFalse,
 			inputAthenzDomain:   onboardedAthenzDomain,
 			existingAuthzPolicy: nil,
 			expectedAuthzPolicy: getExpectedAuthzPolicy(),
 			item:                Item{Operation: model.EventUpdate, Resource: onboardedService},
 		},
+		{
+			name:                "create empty Authorization Policy spec when there is create event of service which doesn't have roles / policies defined",
```

Comment:
condition is: doesn't have both roles and policies defined.
When there are roles but no policies, or when there are policies but no roles. Will the conversion behave correctly? Do you think we can add these two conditions as test cases as well?

Reply:
Sure, added test cases to cover all the different scenarios.